editor/animation_track_editor.cpp

```diff
@@ -4079,6 +4079,22 @@ int AnimationTrackEditor::_confirm_insert(InsertData p_id, int p_last_track, boo
 
 	undo_redo->commit_action();
 
+	if (created && String(p_id.path) == "Sprite:frame") {
```

Comment:
This kind of string comparison is likely to be unreliable, does it work if the Sprite is not named "Sprite"?

This is not the proper way to do this, you should instead rely on the `PROPERTY_USAGE_KEYING_INCREMENTS` usage flag which is assigned for this property, and which is used already in `editor_inspector.cpp` to enable the increment of the frame number on keying.

This logic should also be done earlier, here you're after `undo_redo->commit_action()`, so this breaks the undo/redo feature.

Reply:
Thank you for the feedback this is my first time doing something like this sorry for the mistakes. 

I will try changing the things you mentioned. :)

Reply:
No worry, you did a good job already finding a partial solution :)

You can look a bit further up in the section with the comment "// Hack.", I think you might need to do similar handling for this case (retrieving the property to get its usage flag, checking that it's `PROPERTY_USAGE_KEYING_INCREMENTS`, and if so increment the position by the timeline step).